docs/computer/student_Reports/Acoustic Source - Mohd Hozaifa Khan.md

```diff
@@ -0,0 +1,161 @@
+
+The possible answers to the Questions _"Which side does the source S lie with respect to O, M1, M2?"_ can be **right** or **left**. A more accurate answer could the angle in degrees or radians subtended by source S at O from Line M1,M2. *Determining this angle in essence is a rudimentary form of Sound Source Localization*.
+
+Due to the unsymmetrical arrangement of Source, there will be a phase difference and hence a path difference in the waves recieved by M1 and M2. Calculating this path differennce can help calcuate the angle.
+
+Practically, instead of calculating path difference, it is prefered to determine Time difference between the arrivals of sound at the two detectors or micrphones. Hence, the technique is called *Time Difference of Arrival* method or **TDOA**
+
+Most of the developement in this technique has been on improving the accuracy of calculation of Time Difference of Arrival. It has been either Hardware based improvement or more accurate Mathematical approximation to the process of determining TDOA. The mathematical improvements have been fcoused on Cross-correllation is performed. The most famous being GCC, GCC-PHAT and SRP GCC-PHAT.
+
+---
+
+## Technologies Used
+
+**Language**: Python 3.7
+
+**Libraries**:
+
++ PyAudio  0.2.11
++ SciPy 1.4.1
++ NumPy 1.18.1
++ LibROSA 0.7.2
++ Matplotlib 3.2.1
+
+**Technique** : TDOA (Multi-lateration)
+
+**Hardware**:
+
++ 2 Microphones ( Samsung in-ear Headphones )
++ 2 Audio to USB convertors
++ Laptop
+
+---
+
+## Methodology
+
+We have simply tried to determine the direction of arrival with a resolution of **90 degrees!!**. In simple terms : _left or right_.
+
+We haven't gone with the sophisticated way of calculating TDOA or imporving the accuracy of Cross correllation using GCC (Generalised Cross-Correlaltion). We simply determined the sign of Rxy (Cross Correllation Coefficient) and determined the orientation.
+
+We didn't try it practcally, but a simple ArgMAX of Rxy gives the TDOA, which can be used for Angle Calculation.
+
+## Results and Performance
+
+Work is incomplete to mention any result.
+
+Resolution: not Measured Yet.
+
+Accuracy: Works in some cases.
+
+---
+
+## Problems faced
+
+### Lack of Audio jack ports
+
+First problem was the lack of no. of input *AUDIO ports*. Most of the Microphones currently support AUDIO jack as input to system. There are mostly single USB jack port available in a PC, Board, etc. It was very difficult to recieve acoustic data from 2 diffrent devices at the same time.
+
+#### _Our Solution_
+
+We used audio jack to USB convertors since there are multiple USB ports available in most of devices. Also, PyAudio provides support to create two independant streams to listen to different audio ports. Hence, we were able to collect and process data from two different microphones simultaneously.
```

Comment:
```suggestion
We used the audio jack to USB convertors since there are multiple USB ports available in most of the devices. Also, PyAudio provides support to create two independent streams to listen to different audio ports. Hence, we were able to collect and process data from two different microphones simultaneously.
```